src/plugins/vis_type_table_new/public/components/table_vis_component.tsx

```diff
@@ -38,7 +38,7 @@ export const TableVisComponent = ({
   const pagination = usePagination(visConfig, rows.length);
 
   const sortedRows = useMemo(() => {
-    return uiState.sort?.colIndex && uiState.sort.direction
+    return uiState.sort.colIndex !== null && uiState.sort.direction
       ? orderBy(rows, columns[uiState.sort.colIndex]?.id, uiState.sort.direction)
       : rows;
```

Comment:
```suggestion
    return columns[uiState.sort?.colIndex]?.id
      ? orderBy(rows, columns[uiState.sort.colIndex].id, uiState.sort.direction || <whatever the default is>)
      : rows;
```

Reply:
@ananzh confirmed that this is not directly user input.

Reply:
I am still not comfortable that we are not checking the validity of `id` before passing it to `orderBy`. 


PS, I would have preferred for lodash not even be used here.